### PR TITLE
Adding reproducible benchmarks

### PR DESCRIPTION
**Pull request by kunalgosar**
Adds in reproducible benchmarking scripts.

### CODE REVIEW

File: benchmarks/plots/log_parser.ipynb [hunk 1]
```diff
@@ -0,0 +1,268 @@
+{
```
Comment (devin-petersohn) [inline benchmarks/plots/log_parser.ipynb]: Is this needed in the repo?

File: modin/pandas/indexing.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
 import ray
 from warnings import warn
 
-from .utils import (_get_nan_block_id, extractor,
-                    _mask_block_partitions, writer, _blocks_to_series)
+from .utils import (_get_nan_block_id, extractor, _mask_block_partitions,
```
Comment (devin-petersohn) [inline modin/pandas/indexing.py]: I think this change and the change in `utils` should go in a separate PR.

@simon-mo You made these changes would you create that PR?

File: benchmarks/plots/log_parser.ipynb (hunk 1 above)
Comment (kunalgosar) [reply benchmarks/plots/log_parser.ipynb]: I can remove it, if that is better.

File: modin/pandas/indexing.py (hunk 2 above)
Comment (kunalgosar) [reply modin/pandas/indexing.py]: I've reverted these changes
